pennylane_lightning/core/bindings/Bindings_nb.hpp

```diff
@@ -0,0 +1,794 @@
+ * nb::c_contig> Array with probabilities in numpy format
+ */
+template <class StateVectorT>
+nb::ndarray<typename StateVectorT::PrecisionT, nb::numpy, nb::c_contig>
+probsForWires(Measurements<StateVectorT> &M,
+              const std::vector<std::size_t> &wires) {
+    using PrecisionT = typename StateVectorT::PrecisionT;
+    auto probs_vec = M.probs(wires);
+    return createArrayFromVector<PrecisionT>(probs_vec);
+}
+
+/**
+ * @brief Register probs method for all wires with proper data ownership
+ *
+ * @tparam StateVectorT State vector type
+ * @param M Measurements object
+ * @return nb::ndarray<typename StateVectorT::PrecisionT, nb::numpy,
+ * nb::c_contig> Array with probabilities in numpy format
+ */
+template <class StateVectorT>
+nb::ndarray<typename StateVectorT::PrecisionT, nb::numpy, nb::c_contig>
+probsForAllWires(Measurements<StateVectorT> &M) {
+    using PrecisionT = typename StateVectorT::PrecisionT;
+    auto probs_vec = M.probs();
+    return createArrayFromVector<PrecisionT>(probs_vec);
+}
+
+/**
+ * @brief Generate samples with proper data ownership
+ *
+ * @tparam StateVectorT State vector type
+ * @param M Measurements object
+ * @param num_wires Number of wires
+ * @param num_shots Number of shots
+ * @return nb::ndarray<std::size_t, nb::numpy, nb::c_contig> 2D array with
+ * samples in numpy format
+ */
+template <class StateVectorT>
+nb::ndarray<std::size_t, nb::numpy, nb::c_contig>
+generateSamples(Measurements<StateVectorT> &M, std::size_t num_wires,
+                std::size_t num_shots) {
+    auto result = M.generate_samples(num_shots);
+    return create2DArrayFromVector<std::size_t>(result, num_shots, num_wires);
+}
+
+/**
+ * @brief Register backend-agnostic measurement class functionalities.
+ *
+ * @tparam StateVectorT
+ * @param pyclass Nanobind's class object to bind measurements
+ */
+template <class StateVectorT, class PyClass>
+void registerBackendAgnosticMeasurements(PyClass &pyclass) {
+    // Add probs method for specific wires
+    pyclass.def("probs", &probsForWires<StateVectorT>,
+                "Calculate probabilities for specific wires.");
+
+    // Add probs method for all wires
+    pyclass.def("probs", &probsForAllWires<StateVectorT>,
+                "Calculate probabilities for all wires.");
+
+    // Add expval method for observable
+    pyclass.def(
+        "expval",
+        [](Measurements<StateVectorT> &M, const Observable<StateVectorT> &obs) {
+            return M.expval(obs);
+        },
+        "Calculate expectation value for an observable.");
+
+    // Add var method for observable
+    pyclass.def(
+        "var",
+        [](Measurements<StateVectorT> &M, const Observable<StateVectorT> &obs) {
+            return M.var(obs);
+        },
+        "Calculate variance for an observable.");
+
+    // Add generate_samples method
+    pyclass.def("generate_samples", &generateSamples<StateVectorT>,
+                "Generate samples for all wires.");
+}
```

Comment:
Need to add binding for newly introduced `set_random_seed`: e.g. https://github.com/PennyLaneAI/pennylane-lightning/commit/eaf73caa8737460865c282312110bf4e0dcb28e9#diff-d3a839f5744662a4c49119085cd37a09ef0bd7c0ed9bc2a21b2789994b4d2338R496